### PR TITLE
Patient name disappears

### LINKED ISSUE #439: Patient name disappears
The patient name is shown as the suffix of the breadcrumb - but if you change tabs, then the URL is updated and the name is lost.

To reproduce
- navigate to the patient list, click on a patient
- observe that the breadcrumb shows `Patients / Patient Name`
- click on the programs tab
- the breadcrumb now shows `Patients / [patient id]`

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #439 

Fixed the issue with the breadcrumb suffix - which is because the effect was running off the full `location` object, which includes the search parameters.

Couple of small offroads.. 
- bringing the new tab component in to the patient detail view, which reduces the code in that component and creates some merge conflicts
- adding a tooltip to the patient creation list

Another thing which I'd noticed when looking at these: the patient filter is Case Sensitive, which means that if I search for `tina` I don't find anyone. But there is a `Tina` hiding there.. can I update the query to resolve that? I would expect users to type in lowercase

### CODE REVIEW

File: client/packages/common/src/hooks/useBreadcrumbs/useBreadcrumbs.tsx [hunk 1]
```diff
@@ -47,7 +47,7 @@ export const useBreadcrumbs = (topLevelPaths: string[] = []) => {
     }, '');
     setUrlParts(urlParts);
     setSuffix(undefined);
-  }, [location]);
+  }, [pathname]);
```
Comment (clemens-msupply) [inline client/packages/common/src/hooks/useBreadcrumbs/useBreadcrumbs.tsx:50]: There is a react linter rule:
`"react-hooks/exhaustive-deps": "warn" // Checks effect dependencies`

https://reactjs.org/docs/hooks-rules.html

wounder if we should enable it, it usually quite easy to miss a dependency...

**Review by clemens-msupply — APPROVED** [review]
Looks good thanks, yes please feel free to update the query to be case insensitive

Sorry about the merge conflicts

File: client/packages/common/src/hooks/useBreadcrumbs/useBreadcrumbs.tsx (hunk 1 above)
Comment (mark-prins) [reply client/packages/common/src/hooks/useBreadcrumbs/useBreadcrumbs.tsx:50]: I mostly find that one annoying 😁 Often there's a good reason ( I tell myself ) for not including a dependency which, strictly speaking, should be included. 
Maybe though..

Comment (clemens-msupply) [reply client/packages/common/src/hooks/useBreadcrumbs/useBreadcrumbs.tsx:50]: Thought the same when I first used it but then I was happy when it found some potential issues...

**Pull request merged**